File: src/MixNetFx/ADSundries/ADClassReflector.vb
Imports System.Collections.ObjectModel
Imports System.Reflection

Public Module ADClassReflector

    Function GetProperties(ByVal classInstance As Object) As ADPropertyInfoCollection
        Dim locPropertyInfos() As PropertyInfo = classInstance.GetType.GetProperties
        If locPropertyInfos.Length = 0 Then Return Nothing
        Dim retPropertyInfoCollection As New ADPropertyInfoCollection()
        For Each locPropertyInfo As PropertyInfo In locPropertyInfos
            retPropertyInfoCollection.Add(locPropertyInfo)
        Next
        Return retPropertyInfoCollection
    End Function

End Module

Public Class ADPropertyInfoCollection
    Inherits KeyedCollection(Of String, PropertyInfo)

    Protected Overrides Function GetKeyForItem(ByVal item As System.Reflection.PropertyInfo) As String
        Return item.Name
    End Function
End Class

File: src/NetFxCSharp/ADSundries/ADClassReflector.cs
using System.Collections.ObjectModel;
using System.Reflection;

namespace ActiveDev
{

    public static class ADClassReflector
    {

        public static ADPropertyInfoCollection GetProperties(object classInstance)
        {
            PropertyInfo[] locPropertyInfos = classInstance.GetType().GetProperties();
            if (locPropertyInfos.Length == 0)
                return null;
            var retPropertyInfoCollection = new ADPropertyInfoCollection();
            foreach (PropertyInfo locPropertyInfo in locPropertyInfos)
                retPropertyInfoCollection.Add(locPropertyInfo);
            return retPropertyInfoCollection;
        }

    }

    public class ADPropertyInfoCollection : KeyedCollection<string, PropertyInfo>
    {

        protected override string GetKeyForItem(PropertyInfo item)
        {
            return item.Name;
        }
    }
}